test/e2e_node/docker_util.go

```diff
@@ -60,3 +64,108 @@ func isDockerNoNewPrivilegesSupported() (bool, error) {
 	}
 	return version.GTE(semver.MustParse("1.23.0")), nil
 }
+
+// isDockerLiveRestoreSupported returns true if live-restore is supported in
+// the current Docker version.
+func isDockerLiveRestoreSupported() (bool, error) {
+	version, err := getDockerAPIVersion()
+	if err != nil {
+		return false, err
+	}
+	return version.GTE(semver.MustParse("1.26.0")), nil
+}
+
+// isDockerLiveRestoreEnabled returns true if live-restore is enabled in the
+// Docker.
+func isDockerLiveRestoreEnabled() (bool, error) {
+	c, err := client.NewClient(defaultDockerEndpoint, "", nil, nil)
+	if err != nil {
+		return false, fmt.Errorf("failed to create docker client: %v", err)
+	}
+	info, err := c.Info(context.Background())
+	if err != nil {
+		return false, fmt.Errorf("failed to get docker info: %v", err)
+	}
+	return info.LiveRestoreEnabled, nil
+}
+
+// setDockerLiveRestore sets the live-restore config to the specified value and
+// then restarts the Docker daemon.
+func setDockerLiveRestore(value bool) error {
+	config, err := getDockerDaemonConfig()
+	if err != nil {
+		return err
+	}
+	config["live-restore"] = value
+	if err := setDockerDaemonConfig(config); err != nil {
+		return err
+	}
+	return restartDockerDaemon()
+}
+
+// stopDockerDaemon starts the Docker daemon.
+func startDockerDaemon() error {
+	switch {
+	case isSystemd():
+		_, err := runCommand("systemctl", "start", "docker")
+		return err
+	default:
+		_, err := runCommand("service", "docker", "start")
+		return err
+	}
+}
+
+// stopDockerDaemon stops the Docker daemon.
+func stopDockerDaemon() error {
+	switch {
+	case isSystemd():
+		_, err := runCommand("systemctl", "stop", "docker")
+		return err
+	default:
+		_, err := runCommand("service", "docker", "stop")
+		return err
+	}
+}
+
+// restartDockerDaemon restarts the Docker daemon.
+func restartDockerDaemon() error {
+	switch {
+	case isSystemd():
+		_, err := runCommand("systemctl", "restart", "docker")
+		return err
+	default:
+		_, err := runCommand("service", "docker", "restart")
+		return err
+	}
+}
+
+// getDockerDaemonConfig returns the deserialized Docker daemon config from the
+// 'dockerDaemonConfigName'.
+func getDockerDaemonConfig() (map[string]interface{}, error) {
+	if _, err := os.Stat(dockerDaemonConfigName); os.IsNotExist(err) {
+		return make(map[string]interface{}), nil
+	}
+	data, err := ioutil.ReadFile(dockerDaemonConfigName)
+	if err != nil {
+		return nil, err
+	}
+	var config interface{}
+	err = json.Unmarshal(data, &config)
+	if err != nil {
+		return nil, err
+	}
+	return config.(map[string]interface{}), nil
+}
+
+// setDockerDaemonConfig writes the serialized Docker daemon config to the
+// configuration file 'dockerDaemonConfigName'.
+func setDockerDaemonConfig(config map[string]interface{}) error {
```

Comment:
I don't think you need to build a new image. I think you can just add another image [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/jenkins/image-config.yaml), and modify the image init script: https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/jenkins/gci-init.yaml (or maybe just do this for the existing image, if you don't need to test with it off).